plsql-frontend/src/main/java/org/sonar/plugins/plsqlopen/api/DmlGrammar.java

```diff
@@ -287,9 +304,10 @@ private static void createSelectExpression(LexerfulGrammarBuilder b) {
                             b.optional(HAVING_CLAUSE),
                             b.optional(HIERARCHICAL_QUERY_CLAUSE),
                             b.optional(b.firstOf(
-                                    b.sequence(ORDER_BY_CLAUSE,  b.optional(FOR_UPDATE_CLAUSE)), 
+                                    b.sequence(ORDER_BY_CLAUSE, b.optional(FOR_UPDATE_CLAUSE)),                                     
+                                    b.sequence(ORDER_BY_CLAUSE, b.optional(ROW_LIMITING_CLAUSE)), 
```

Comment:
As it is now, when the parser is evaluating something like `ORDER BY id FETCH FIRST 5 ROWS ONLY`, it will match the first option (ORDER_BY_CLAUSE, optional FOR_UPDATE_CLAUSE) and the row limiting clause will never be reached.

To work correctly, this should be a `b.optional(b.firstOf(FOR_UPDATE_CLAUSE, ROW_LIMITING_CLAUSE))`, instead of two different lines.